data/model/repository.py

```diff
@@ -526,16 +533,26 @@ def _get_sorted_matching_repositories(
             .order_by(SQL("score").desc(), RepositorySearchScore.id)
         )
 
+        # If an organization/user was found and it was not blank.
+        if (user_name_value is not None) and (user_name_value != ""):
+            query = (
+                query.switch(Repository)
+                .join(Namespace, on=(Namespace.id == Repository.namespace_user))
+                .where(Namespace.username == user_name_value)
+            )
+            is_namespace_table_joined = True
+
     if repo_kind is not None:
         query = query.where(Repository.kind == Repository.kind.get_id(repo_kind))
 
     if not include_private:
         query = query.where(Repository.visibility == _basequery.get_public_repo_visibility())
 
     if not ids_only:
-        query = query.switch(Repository).join(
-            Namespace, on=(Namespace.id == Repository.namespace_user)
-        )
+        if not is_namespace_table_joined:
```

Comment:
I'll change this. Just want to mention that I did it this way just in case future logic needed to be added that was relevant only for `if not ids_only`

Reply:
understood, i think it's fine to combine them.

Reply:
Addressed in 1e7050b8602c7605b55de20b6f0e4833e37ddfa3. Given the previous comment, I also removed the `on=(Namespace.id == Repository.namespace_user)`.